acmens.py

```diff
@@ -200,42 +202,56 @@ def _poll_until_not(url, pending_statuses, err_msg):
         order["authorizations"][0], None, "Error getting challenges"
     )
 
-    challenge = [c for c in chl_result["challenges"] if c["type"] == "http-01"][0]
+    type_id = 'dns-01' if challenge_type == 'dns' else 'http-01'
+    challenge = [c for c in chl_result["challenges"] if c["type"] == type_id][0]
     token = re.sub(r"[^A-Za-z0-9_\-]", "_", challenge["token"])
     keyauthorization = "{0}.{1}".format(challenge["token"], thumbprint)
 
     # build request for the server to test this challenge.
     test_url = challenge["url"]
     test_raw = "{}"
 
-    # challenge response for server
-    response = {
-        "uri": ".well-known/acme-challenge/{0}".format(challenge["token"]),
-        "data": keyauthorization,
-    }
-
     # Step 6: Ask the user to host the token on their server
-    sys.stderr.write(
+    if challenge_type == 'dns':
+        sys.stderr.write(
+        """\
+Please update your DNS to have the following TXT record:
+
+--------------
+DNS TXT record: _acme-challenge.{domain}
```

Comment:
I think it would be a good idea to format it like a bind9 TXT record:

```
_acme-challenge    IN    TXT ( \"{keyauth}\" )
```

Also, I don't think the `{domain}` part is needed.